### PR TITLE
platform: etherscan: add support for solc remaps, evm version

### LINKED ISSUE #150: Etherscan: improve multi files support
`crytic-compile` fails on some codebase with multiple files. For example on https://etherscan.io/address/0x640022f8a9f00992fa63ee6bae25c2829deadd2d

```
crytic_compile.platform.exceptions.InvalidCompilation: Invalid solc compilation Error: Source "openzeppelin-solidity/contracts/access/AccessControl.sol" not found: File not found.
 --> contracts/Roles.sol:4:1:
  |
4 | import "openzeppelin-solidity/contracts/access/AccessControl.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Source "openzeppelin-solidity/contracts/access/Ownable.sol" not found: File not found.
 --> contracts/Roles.sol:5:1:
  |
5 | import "openzeppelin-solidity/contracts/access/Ownable.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by elopez**
This adds support for solc remaps and EVM version found in the Etherscan JSON.

Fixes #150, #151, #386, #415

### CODE REVIEW

**Comment by 0xalpharush** [comment]
Will this address https://github.com/crytic/crytic-compile/issues/151 and https://github.com/crytic/crytic-compile/issues/150

**Comment by elopez** [comment]
This also seems to fix the case in #386.

The case in #151 requires passing the EVM target, which this PR doesn't, but it could be added as a follow up.

#150:
* 0x640022f8a9f00992fa63ee6bae25c2829deadd2d works both with and without this PR
* 0x2a311e451491091d2a1d3c43f4f5744bdb4e773a has broken remappings (with absolute paths). This PR rejects them currently
```WARNING:CryticCompile:Remapping @0x/contracts-utils=/home/cluracan/code/0x-protocol/node_modules/@0x/contracts-utils is potentially unsafe, skipping
WARNING:CryticCompile:Remapping @0x/contracts-erc20=/home/cluracan/code/0x-protocol/contracts/zero-ex/node_modules/@0x/contracts-erc20 is potentially unsafe, skipping
```
we could override any absolute paths and make them relative though, that would work in this case from what I can see.
* 0xcE4a49d7ed99C7c8746B713EE2f0C9aA631688d8 works both with and without this PR

**Comment by elopez** [comment]
I added some more code to handle those two cases 👍

File: crytic_compile/platform/etherscan.py [hunk 1]
```diff
@@ -432,13 +443,44 @@ def _convert_version(version: str) -> str:
...
+
+        # if path is absolute, relativize it
+        if dest.is_absolute():
+            dest = PurePosixPath(*dest.parts[1:])
+
+        dest_disk = Path(allowed_directory, dest)
+
+        if os.path.commonpath((allowed_path, os.path.abspath(dest_disk))) != allowed_path:
+            LOGGER.warning("Remapping %s=%s is potentially unsafe, skipping", origin, dest)
+            continue
```
Comment (0xalpharush) [inline crytic_compile/platform/etherscan.py:481]: Not blocking but we should extract this into utils so we can have a common API to validate absolute path resolution (ref https://github.com/crytic/crytic-compile/commit/8a7c037261ff618964c23cab9f401413bb191783)

**Pull request merged**